spec/checkPullRequestLabelsSpec.js

```diff
@@ -0,0 +1,269 @@
+      payload.payload.pull_request.user.login = 'kevintab95';
+      spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+      await robot.receive(payload);
+
+      expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      expect(github.issues.addAssignees).not.toHaveBeenCalled();
+    });
+
+    it('does not assign if another label got added', async () => {
+      const label = {
+        id: 638839900,
+        node_id: 'MDU6TGFiZWw2Mzg4Mzk5MDA=',
+        url: 'https://api.github.com/repos/oppia/oppia/labels/PR:%20released',
+        name: 'dependencies',
+        color: '00FF00',
+      };
+      // Set the payload action and label which will simulate adding
+      // the changelog label.
+      payload.payload.action = 'labeled';
+      payload.payload.label = label;
+      spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+      await robot.receive(payload);
+
+      expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      expect(github.issues.addAssignees).not.toHaveBeenCalled();
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+
+    it('should not assign when invalid changelog is applied', async () => {
+      const label = {
+        id: 638839900,
+        node_id: 'MDU6TGFiZWw2Mzg4Mzk5MDA=',
+        url: 'https://api.github.com/repos/oppia/oppia/labels/PR:%20released',
+        name: 'PR CHANGELOGS: Server Errors -- @kevintab95',
+        color: '00FF00',
+      };
+
+      payload.payload.action = 'labeled';
+      payload.payload.label = label;
+      spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+      await robot.receive(payload);
+
+      expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      expect(github.issues.addAssignees).not.toHaveBeenCalled();
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+
+    it('should not assign when there are review comments', async () => {
+      const label = {
+        id: 638839900,
+        node_id: 'MDU6TGFiZWw2Mzg4Mzk5MDA=',
+        url: 'https://api.github.com/repos/oppia/oppia/labels/PR:%20released',
+        name: 'PR CHANGELOG: Server Errors -- @kevintab95',
+        color: '00FF00',
+      };
+
+      payload.payload.action = 'labeled';
+      payload.payload.label = label;
+      // Simulate when the payload alread has review comments.
+      payload.payload.pull_request.review_comments = 2;
+      spyOn(checkPullRequestLabelModule, 'checkAssignee').and.callThrough();
+      await robot.receive(payload);
+
+      expect(checkPullRequestLabelModule.checkAssignee).toHaveBeenCalled();
+      expect(github.issues.addAssignees).not.toHaveBeenCalled();
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('when pull request gets opened or reopened', () => {
+    it('pings pr author when there is no changelog label', async () => {
```

Comment:
One more case - does not ping author if there is a label.

Reply:
I don't think it is possible to add a label more than once.

Reply:
I meant a case where PR is reopened and there is a changelog label - in that case, there should be no ping to the author, right?

Reply:
Ohh I get.
Added.